packages/common/src/core/behaviors/IWithCardBehavior.ts

```diff
@@ -0,0 +1,15 @@
+import { IHubCardViewModel } from "../types/IHubCardViewModel";
+
+export interface IWithCardBehavior {
+  /**
+   * Convert an entity into a card view model that can
+   * be consumed by the suite of hub gallery components
+   *
+   * @param target card link contextual target
+   * @param locale internationalization locale
+   */
+  convertToCardViewModel(
+    target: "ago" | "view" | "workspace",
```

Comment:
Will the Entity class have access to the AGO home url? Or will we need to pass this in? Also remember that the `Site` entities `view` target will need to visit the site rather than a content view.

Reply:
This class method ultimately delegates out to the `getCardViewModelFromProjectEntity` function which constructs the card model's `titleUrl` based on the `target` supplied. To do this, we're using the same functions used to "enrich" a project's search result with links...so for `target` = "ago", we use the `getItemHomeUrl` which constructs the AGO home url, so I don't think we need to pass that in. Let me know if that makes sense though.  

r.e. `Site` entities, yes, that's a valid point, but I don't necessarily think that's a concern here since the `Site` entity will similarly define its own `getCardViewModelFromSiteEntity` function which can handle constructing a unique `view` url.